src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/modelchecking/ModelCheckingStrategy.kt

```diff
@@ -537,33 +537,59 @@ internal class ModelCheckingMonitorTracker : MonitorTracker {
 
 class ModelCheckingParkingTracker(val allowSpuriousWakeUps: Boolean = false) : ParkingTracker {
 
-    private val parked = mutableThreadMapOf<Boolean>()
+    private object PERMIT
+
+    private enum class State {
+        UNPARKED, PARKED, INTERRUPTED,
```

Comment:
let's add an additional state instead of adding permit and draw a state diagram for clarity